dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftClusterUsageMetricsTaskTest.java

```diff
@@ -0,0 +1,152 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.redshift;
+
+import static com.google.edwmigration.dumper.application.dumper.connector.redshift.RedshiftClusterUsageMetricsTask.MetricConfig;
+import static com.google.edwmigration.dumper.application.dumper.connector.redshift.RedshiftClusterUsageMetricsTask.MetricName;
+import static com.google.edwmigration.dumper.application.dumper.connector.redshift.RedshiftClusterUsageMetricsTask.MetricType;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import autovalue.shaded.com.google.common.collect.ImmutableList;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.model.Datapoint;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.GetMetricStatisticsRequest;
+import com.amazonaws.services.cloudwatch.model.GetMetricStatisticsResult;
+import com.amazonaws.services.redshift.AmazonRedshift;
+import com.amazonaws.services.redshift.model.Cluster;
+import com.amazonaws.services.redshift.model.DescribeClustersResult;
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedInterval;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractTaskTest;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RedshiftRawLogsDumpFormat;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public class RedshiftClusterUsageMetricsTaskTest extends AbstractTaskTest {
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Mock private AmazonRedshift redshiftClientMock;
+
+  @Mock private AmazonCloudWatch cloudWatchClientMock;
+
+  private static final ImmutableList<Cluster> TEST_CLUSTERS =
+      ImmutableList.of(
+          new Cluster().withClusterIdentifier("clId1"),
+          new Cluster().withClusterIdentifier("clId2"));
+
+  private static final ZonedDateTime CURR_DATE_TIME =
+      ZonedDateTime.of(2024, 01, 01, 11, 0, 0, 0, ZoneId.of("UTC"));
+  private static final ZonedInterval TEST_INTERVAL =
+      new ZonedInterval(
+          ZonedDateTime.of(2024, 01, 01, 10, 0, 0, 0, ZoneId.of("UTC")),
+          ZonedDateTime.of(2024, 01, 01, 10, 10, 0, 0, ZoneId.of("UTC")));
```

Comment:
Prefer using different numbers for different timestamp components. e.g. 2024, 01, 02, 10, 11, 12 ...... for the test values.